numpy/core/src/multiarray/methods.c

```diff
@@ -1918,10 +1918,30 @@ array_setstate(PyArrayObject *self, PyObject *args)
     Py_XDECREF(PyArray_DESCR(self));
     fa->descr = typecode;
     Py_INCREF(typecode);
-    nd = PyArray_IntpFromSequence(shape, dimensions, NPY_MAXDIMS);
-    if (nd < 0) {
+    nd = PyTuple_GET_SIZE(shape);
+    if (nd > NPY_MAXDIMS) {
+        PyErr_Format(PyExc_ValueError, "Too many dimensions");
```

Comment:
I am not sure I follow this change. Why not use IntpFromSequence and just raise the error if it returns something to short? Likely, you have to pass `NPY_MAXDIMS + 1`, but that seems OK?

Reply:
Dang, sorry for not reading the comment carefully (missed it after a first quick nitpick comment, I guess)...